src/index.js

```diff
@@ -664,31 +662,45 @@ class ReactTooltip extends React.Component {
     clearInterval(this.intervalUpdateContent);
   }
 
+  hasCustomColors() {
+    return Boolean(
+      Object.keys(this.state.customColors).find(color => color !== "border" && this.state.customColors[color]) ||
+      (this.state.border && this.state.customColors["border"])
+    );
+  }
+
   render() {
     const { extraClass, html, ariaProps, disable } = this.state;
     const placeholder = this.getTooltipContent();
     const isEmptyTip = this.isEmptyTip(placeholder);
-    const tooltipClass =
-      "__react_component_tooltip" +
-      (this.state.show && !disable && !isEmptyTip ? " show" : "") +
-      (this.state.border ? " border" : "") +
-      ` place-${this.state.place}` + // top, bottom, left, right
-      ` type-${this.state.type}` + // dark, success, warning, error, info, light
-      (this.props.delayUpdate ? " allow_hover" : "") +
-      (this.props.clickable ? " allow_click" : "");
+
+    const tooltipClass = classname(
+      "__react_component_tooltip",
+      { "show": this.state.show && !disable && !isEmptyTip },
+      { "border": this.state.border },
+      { "place-top": this.state.place === "top" },
+      { "place-bottom": this.state.place === "bottom" },
+      { "place-left": this.state.place === "left" },
+      { "place-right": this.state.place === "right" },
+      { ["type-" + (this.hasCustomColors() ? "custom" : this.state.type)]: this.state.type },
+      { "allow_hover": this.props.delayUpdate },
+      { "allow_click": this.props.clickable }
+    );
+
```

Comment:
you could replace classnames use by the actual implementation in master

Reply:
Refactored according to the **master** branch implementation.